modules/fbs/includes/fbs.loan.inc

```diff
@@ -10,61 +10,82 @@
  *
  * @param object $account
  *   User to fetch list for.
+ * @param bool $reset
+ *   If TRUE cache is reset (default: FALSE)
  *
  * @return array
  *   Loan information.
  */
-function fbs_loan_list($account) {
-  $res = array();
-  try {
-    $res = fbs_service()->MaterialLoans->getLoans(fbs_service()->agencyId, fbs_patron_id($account));
-  }
-  catch (Exception $e) {
-    watchdog_exception('fbs', $e);
-  }
+function fbs_loan_list($account, $reset = FALSE) {
+  $results = &drupal_static(__FUNCTION__, array());
 
-  $ids = array();
-  foreach ($res as $loan) {
-    $ids[$loan->loanDetails->recordId] = $loan->loanDetails->recordId;
-  }
-  $ids = ding_provider_build_entity_id($ids);
+  if (empty($results) || $reset) {
+    // Check if ding_session_cache is available.
+    if (module_exists('ding_session_cache') && !$reset) {
+      $results = ding_session_cache_get('fbs', 'loans', array());
+    }
 
-  $result = array();
-  foreach ($res as $loan) {
-    $id = $loan->loanDetails->loanId;
-
-    $loan_data = array(
-      'ding_entity_id' => $ids[$loan->loanDetails->recordId],
-      'loan_date' => $loan->loanDetails->loanDate,
-      'expiry' => $loan->loanDetails->dueDate,
-      'renewable' => $loan->isRenewable,
-      'materials_number' => $loan->loanDetails->materialItemNumber,
-    );
-
-    // If this is a periodical, add in issue data.
-    if (isset($loan->loanDetails->periodical)) {
-      $periodical = $loan->loanDetails->periodical;
-      $vol = $periodical->volume;
-      if (!empty($periodical->volumeNumber)) {
-        $vol .= '.' . $periodical->volumeNumber;
+    // If both static cache and session cache failed, try reloading information
+    // from FBS.
+    if (empty($results) || $reset) {
```

Comment:
See comment above regarding handling of empty results from FBS.

Reply:
Changed default value to NULL and using is_null to check it.